packages/guides/src/Compiler/Passes/AutomaticMenuPass.php

```diff
@@ -53,9 +64,46 @@ public function run(array $documents, CompilerContextInterface $compilerContext)
                 continue;
             }
 
-            $documentEntry = $projectNode->getDocumentEntry($documentNode->getFilePath());
-            $documentEntry->setParent($rootDocumentEntry);
-            $rootDocumentEntry->addChild($documentEntry);
+            $filePath = $documentNode->getFilePath();
+            $pathParts = explode('/', $filePath);
+            $documentEntry = $projectNode->getDocumentEntry($filePath);
+            if (count($pathParts) === 1 || count($pathParts) === 2 && in_array($pathParts[1], $indexNames, true)) {
+                $documentEntry->setParent($rootDocumentEntry);
+                $rootDocumentEntry->addChild($documentEntry);
+                continue;
+            }
+
+            $fileName = array_pop($pathParts);
+            $path = implode('/', $pathParts);
+            if (in_array($fileName, $indexNames, true)) {
+                array_pop($pathParts);
+                $path = implode('/', $pathParts);
+            }
+
+            $parentFound = false;
+            foreach ($indexNames as $indexName) {
+                $indexFile = $path . '/' . $indexName;
+                try {
+                    $parentEntry = $projectNode->getDocumentEntry($indexFile);
```

Comment:
```suggestion
                    $parentEntry = $projectNode->findDocumentEntry($indexFile);
                    if ($parentEntry === null) {
                       continue;
                    }
```
This seems better to me as you are not using exceptions for situations that are not exceptional

Reply:
Done